### ISSUE TITLE
Request: Add zoom function to image editor

### ISSUE THREAD

**Issue by Freekers** [ShareX/ShareX#2325]
Currently, the image editor does not have a zoom function, which makes it hard to edit small images or precisely crop etc. Thank you!

**Comment by edphemeria** [comment]
+1, I sometimes annotate very large files which require lots of scrolling.
Huge thanks for ShareX, it's just the best!

**Comment by Davexx** [comment]
Yes, I have 2 monitors and sometimes I cannot access portions of the screen capture.  I would love this feature as well.  and thanks for the awesome tool.  It's very thorough!

**Comment by nepomuc** [comment]
Yes, please add this feature! The image editor is so feature-rich, it even has two different ways to anonymize regions by blurring or pixelating them. But it has no zoom in/out.

**Comment by TJohn7002** [comment]
Vote for this

**Comment by Freekers** [comment]
1 year later and I'm still rooting for this feature ;) Pinging @Jaex

**Comment by marcoscheel** [comment]
+1 for this feature

**Comment by FELITH** [comment]
me too

**Comment by markusd1984** [comment]
👍 +1 I discovered this because the picture I opened was too large. 
Without a zoom function, one can't view the image as a whole and to edit needs to scroll around :)

**Comment by viper25** [comment]
Yes, this is one basic feature missing from such an advanced snapshot tool. I was using CTRL + mouse scroll  to zoom in/out instinctively and was mildly surprised to see one couldn't zoom.

**Comment by Jaex** [comment]
Adding zoom is not basic at all.

**Comment by Davexx** [comment]
@Jaex  I believe @viper25 meant "fundamental" when he used the work "basic".  I don't think he was diminishing the effort it would take to add the code.  It's one of those things you quite literally expect to be there especially with such large screen capture real estate these days.

**Comment by Jaex** [comment]
I don't think there is any software with on screen annotation which can zoom. It is not image editor, it is region capture with annotation support.

**Comment by nepomuc** [comment]
@Jaex Do you mean something like this?

![screen_annotation-do_you_mean_something_like_this](https://user-images.githubusercontent.com/864660/40316689-64a37120-5d1f-11e8-8ef3-d008c4306d56.gif)

**Comment by Jaex** [comment]
That is image editor, not on screen annotation. Their region capture don't have zoom and annotation functionality.

**Comment by nepomuc** [comment]
@Jaex Okay, understood. But this issue here is not about the region capture but about the image editor, isn't it? It would be very handy to simply be able to do what's shown in the GIF, but with ShareX. :)

**Comment by Jaex** [comment]
That image editor is actually RegionCaptureForm in codes, modified to act like image editor. So everything I add to region capture on screen annotation also benefits that mode. But this situation makes adding image editor related stuffs difficult. Because it is not coded from beginning to be image editor. So adding such features far more difficult because it is connected to region capture.

**Comment by nepomuc** [comment]
@Jaex I think I just understood a little more what makes it difficult for you to add the zoom feature. From a user's perspective I have to agree with @Davexx . Being able to zoom in the "image editor" is something fundamental, something you expect as a "basic feature". I would love to help in any way but I'm afraid my programming skills won't be very helpful in this case.
You said that it's "modified to act like image editor". So this sounds like potential for giving it the ability to become a real image editor and not only act like one?

**Comment by Jaex** [comment]
That would require making image editor from scratch. But then it would lose benefits of also be able to annotate inside region capture. And because I'm mostly annotating inside region capture. That would be bad for me.

**Comment by Davexx** [comment]
I must restate that I am very appreciative of any free tool and the extreme effort that goes into creating it.  That said, I spent some time in software development and the most successful apps/products were always the ones that listened to the requests of the users and would try to implement those requests as features, because ultimately the users drive the success of any software.

**Comment by Jaex** [comment]
Are you seriously accusing me of not listening the users feedback? Even without any user feedback I was well aware zoom was needed. I also needed it sometimes. How am I gonna implement it? Are you thinking it is easy as saying...

**Comment by nepomuc** [comment]
@Jaex @Davexx I think we can agree on the fact that a zoom feature for the "image editor" would be a very helpful feature. @Jaex You mentioned that you "also needed it sometimes". Do you have some kind of workaround, besides saving the image and opening it in a "real" image editor?

**Comment by Jaex** [comment]
I'm opening image in Paint .NET when I need zoom.

**Comment by nepomuc** [comment]
Alright, that's basically the same I'm doing right now. I would still love to see this feature in the future, since it would be helpful for everyone who voted for it in this thread and also probably for a lot of other users too. But you decide since it's you who would have to do the work. I would be very thankful for the effort. Maybe it helps to think about a solution in a more indirect way, so you don't need to completely rebuild the entire "editor". But you probably know best for yourself how to find a solution. :)

**Comment by viper25** [comment]
> Adding zoom is not basic at all.

@Jaex I didn't mean to trivialize the effort or mean it was simple or basic to implement. As @Davexx said, I meant "fundamental";  I just meant that this tool has almost every other feature I could think of (and more) except zoom which I use when i need to crop large images. It's hard to do so in ShareX.This is an example workflow of what I'm talking about (Photoshop). I use the ALT+Mouse scroll to zoom in /out to the area of interest and THEN crop.

 
![2018-05-22_14-39-26](https://user-images.githubusercontent.com/327990/40346378-2fd5f70a-5dcf-11e8-81ff-b62023228d17.gif)

> So adding such features far more difficult because it is connected to region capture.

The details of the implementation would be upto you. If it's not possible, then so be it. As a user, am just providing a suggestion that would be good to have!

**Comment by vitaliy-kravchenko-dev** [comment]
Completely agree with @viper25
It would be nice to have this functionality inside the editing tool after capturing a screenshot.
As for me, I have 3 monitors and in some situations, it's not easy to select a region.

**Comment by njwgh** [comment]
+1 from me thanks

**Comment by frankydee** [comment]
+1 from me, and I agree with @viper25.

A very limiting side effect of the lack of zoom is that we can't crop images which are bigger than the screen (which is pretty much always the case for scrolling screen captures)

**Comment by derekantrican** [comment]
Agree here. I love using ShareX and all the customizability that it has but the thing I miss most about Snagit (the last screenshot tool I used) was the vast host of image editing tools. The main thing that seems obvious is the ability to zoom in/out on an image that you're editing. With a high res image (eg a screenshot from my phone) I have to do a lot of scrolling to pan around the image. That's a pain (especially if that scrolling includes me trying to adjust a rectangle annotation or something)

**Comment by rodrigotrovao** [comment]
this feature has in every other image software. In this OUTSTANDING tool doesnt?

**Comment by heyvoon** [comment]
Up! This is a BASIC for ANY image editor. UP! UP! UP!
Please implement it.

**Comment by Jaex** [comment]
> this feature has in every other image software. In this OUTSTANDING tool doesnt?

If I made image editor I would also add zoom feature to it from start, it is most fundamental feature of image editors. But ShareX don't have real image editor. It is just region capture with annotation support. How many screen capture software have zoom support in their region capture? Zero?

**Comment by derekantrican** [comment]
@Jaex First off, no one's trying to be hostile and insult you here. We just really want this feature.

Secondly, as I mentioned before, Snagit is a program that is a screenshot tool with advanced annotation and it has this feature. So that's at least 1

**Comment by Jaex** [comment]
SnagIt don't have this feature inside their "region capture". It is inside their image editor as I told. Find me single software which have zoom inside their "region capture".

**Comment by derekantrican** [comment]
That's what we're asking for. We want it in the "image editor"/"image annotator" whatever you want to call it

**Comment by Jaex** [comment]
We don't have "real" image editor. We have region capture in windowed mode instead of fullscreen.

**Comment by derekantrican** [comment]
![image](https://user-images.githubusercontent.com/1558019/56920986-81a25f80-6a79-11e9-86fc-d1990c064b36.png)

This screen. *After* you've done the capture. It doesn't have the ability to zoom in and out on what you've captured. Which makes it difficult to annotate very small captures or if you want to use it to annotate large pictures.

**Comment by njwgh** [comment]
I used Ashampoo snap before this (and paid for it). I way prefer ShareX for all other features and would definitely pay (donate) if it also had the zoom function. Ashampoo snap does have zoom function in region capture screen
![ashampoo2](https://user-images.githubusercontent.com/6837919/56935805-e5d72900-6b36-11e9-9338-1cdeb8b1ea0c.png)
![ashampoo3](https://user-images.githubusercontent.com/6837919/56935811-ef609100-6b36-11e9-90d9-0a3ed2a6c681.png)

**Comment by K0media** [comment]
I think it could be set as an option on the capture/program settings. You could either set to hold `Ctrl` or `Alt` + scrolling the `Mouse Wheel` or even in the touchpad, if it's even necessary.

It would require some more programming if it's implemented along with the on-screen screenshot capture with selected regions though. I believe as an *annotation* or in the *image editor* would fit better for this scenario.

But good idea indeed. Google helped me to reach this issue. I believe this has been requested years ago... =/

**Comment by viper25** [comment]
> But ShareX don't have real image editor. It is just region capture with annotation support. 

I think most commenters here are lost on the difference you are trying to highlight above. As far as users are concerned, once a screenshot it taken, an "image editor" pops up real not not. As it is now, the scrolling capture feature (and cropping other large images) is near unusable without a zoom and crop option.

However that said, thanks for providing this wonderful free tool with great features! Hoping zoom and crop can be added too! :)

**Comment by derekantrican** [comment]
@K0media I would argue that - in the image editor/annotator - scrolling should default to zoom and the current behavior should require the shortcut key (Ctrl, Alt, whatever)

@viper25 crop already exists, just not zoom

**Comment by viper25** [comment]
> @viper25 crop already exists, just not zoom

Yes I was talking about the **zoom and crop** as one feature not 2 separate features.

**Comment by stephen147** [comment]
I would love this feature.

## I like how [FastStone image](https://www.faststone.org/) viewer does it.

In the settings, there are zoom options for at mouse position and Center (at mouse is best imo)
![image](https://user-images.githubusercontent.com/14054664/60336554-94151880-9998-11e9-829e-90912954a29a.png)


## Then here are the context options:
With the zoom at the mouse position enabled and using the <kbd>+</kbd> and <kbd>-</kbd> keys or the mouse wheel, you can zoom in and out to where your mouse is located.

![image](https://user-images.githubusercontent.com/14054664/60336337-f883a800-9997-11e9-9168-99c90d57d8d3.png)

**Comment by syaifulnizamyahya** [comment]
@Jaex Explaining the internals to the user might not be the best course of action. That said, explaining the internals to the developers might be more productive. 

I mean, what outcome would one expect when explaining the internal to the users? Most users(I said most) doesnt really understand and started making speculation(sometimes false, sometimes outright irrelevant) which in the end will exhaust the developers motivation. A simple notes mentioning that the work on it is in queue or we dont have enough resources right now or it requires major refactoring so its well below current priority or else, might be enough. 

As for me, while I dont really understand your situation, I do understand the complication when working with graphics objects(annotations) and bitmaps. I have a rough idea on simple solution but Im not sure if it fits your architecture.

Anyway, I wish the best on sharex and hope that someday we can have zoom feature.

Thanks.

**Comment by Unknow0059** [comment]
I also hope this feature gets implemented.

**Comment by markusd1984** [comment]
Has this been considered? Any Update on this much needed feature?

Scrolling doesn't even help when annotating on larger images, almost tempted to take a screenshot of the image to work around this, lol.

**Comment by nhgiang196** [comment]
Still waiting for this feature. Hope it be completed soon. 
By the way, where is the funding page for shareX? I would love to help =)

**Comment by derekantrican** [comment]
@nhgiang196 Funding page is here: https://getsharex.com/donate/

**Comment by Layarion** [comment]
this is "needed", I'm considering, after years of lightly using shareX, leaving this app behind because what a pain in the @ss this is when you use it for more than 3 seconds at a time.

**Comment by Layarion** [comment]
I've been trying to edit something now for several minutes and the inability to zoom in and see exactly where the border is of the red box I made without mousing over it, is crippling. There's no other way to describe it, this is a horrible experience.

now if I want to continue using this I have to set my desktop resolution to 720p? give up the app? maybe even start here, go to the other app to finish something that otherwise woulda been done better here?

wow.

**Comment by Davexx** [comment]
Windows 10 was updated a while ago to include many quick and useful tools for capturing, marking up, and cropping screen content, all within an interface that allows for zooming large captures and saving the results.  Gone are the clunky days of Windows Snipping Tool.  Here is the new way!

Using the following key combination:

1. **WIN + SHIFT + S**
2. The screen will become shaded and at the top middle of your screen there is an icon menu with the following options:
3. Rectangular Snip, Freeform Snip, Window Snip, FullScreen Snip, and "X" to cancel.
4. Choose one and execute your capture (Rectangular Snip is the default).
5. **Snip Saved to Clipboard** appears in the bottom right of your screen (Notification Center).
6. Click on the Notification, or retrieve it from the Notification Center and click on it there.
7. The Windows **Snip & Sketch app** will appear with your capture result.  You may now mark it up with numerous tools.  There is also a screen capture delay timer (dropdown arrow next to "New Snip").
8. The **Image Crop tool** has a very intuitive ZOOM built-in if you have captured some very large screen real estate.  As you pull in the crop boundary handles, the screen automatically zooms in!

I hope that helps folks who have been looking for this zoom feature in ShareX to know that Windows 10 now has you covered.

**Comment by GuestShark** [comment]
_> 8. The **Image Crop tool** has a very intuitive ZOOM built-in if you have captured some very large screen real estate.  As you pull in the crop boundary handles, the screen automatically zooms in!_

Hi - wanted to add to this that the zoom/crop method you're referencing doesn't seem to work for dual screen captures... when I open it from notification center into Snip & Sketch it doesn't show the full capture, rather it 'opens' zoomed in with no way to zoom back out

**Comment by Davexx** [comment]
Hi, I have Dual screens too.  I just tried it and it worked for me.  Does your interface look like this in Windows 10?
![Snip and sketch](https://user-images.githubusercontent.com/3362914/81355694-80dcda00-908c-11ea-9027-e24a9d52783c.png)
I highlighted the zoom tool (circled, in pink).  Hopefully that works for you, but my dual screen capture appears all zoomed out so I can see both screens in the Snip & Sketch interface.  Al I have to do is use the crop tool and save.

**Comment by GuestShark** [comment]
Thanks for writing back! Must be a versioning issue (I have 10.1806.2112.0) because I do not have that same look with the magnifying glass, rather:
![image](https://user-images.githubusercontent.com/64969079/81420188-95b57e00-911d-11ea-9126-82b3476e6881.png)

**Comment by Davexx** [comment]
Yes, perhaps your Windows 10 needs an update?  
I am using Snip & Sketch 10.1907.2471.0

**Comment by GuestShark** [comment]
> Yes, perhaps your Windows 10 needs an update?
> I am using Snip & Sketch 10.1907.2471.0

Indeed - unfortunately my org controls when those updates are pushed...  I'll report back :)

**Comment by Davexx** [comment]
Well, I am just grateful you got to the bottom of the issue and hopefully it may help someone else in the future who might not be able to get this solution for the screen capture zoom and cropping issue to work.  I am happy it's now possible in 2020, hahaha.

**Comment by nhgiang196** [comment]
This is an idea to zoom out your image. Howerver, you can't undo or zoom in again without losing image pixel : )
![O7R8m98p1v](https://user-images.githubusercontent.com/22267870/83214872-d722d180-a18f-11ea-9c8f-7c8c3cd8808a.gif)

**Comment by viper25** [comment]
This isn't zoom, this is resizing the image. It alters your image while zoom doesn't.

**Comment by rogeriodec** [comment]
I do not believe that after so many years, a simple zoom has not yet been added to the editor ...

**Comment by Jaex** [comment]
> I do not believe that after so many years, a simple zoom has not yet been added to the editor ...

If it is that simple please pull request it. After all project is open source.

**Comment by Davexx** [comment]
@rogeriodec  
See my March 9 2020 comment.  It might help you achieve what you are wanting.

**Comment by beijingjazzpanda** [comment]
The zoom in/out function of FastStone Capture is avaliable in both Capture and Editor, is very handy when review what it captured.. It's suggested to add it if it doesn't need huge effort to develop it.
![7eoHy2QsKb](https://user-images.githubusercontent.com/10892928/90367932-e2eda800-e09b-11ea-95a1-f4d18dd9c538.gif)

**Comment by schousda** [comment]
3.5 years and nothing happend?
I'm waiting for this useful zoom feature too.
Thanks.

**Comment by nhgiang196** [comment]
> 3.5 years and nothing happend?
> I'm waiting for this useful zoom feature too.
> Thanks.

They created magnify instead 🤣🤣
> ![VqC2qhlH6x](https://user-images.githubusercontent.com/22267870/95704445-0e93a580-0c7b-11eb-85ea-4469e1f91434.gif)

This is another "solution" : ))

> This is an idea to zoom out your image. Howerver, you can't undo or zoom in again without losing image pixel : )
> ![O7R8m98p1v](https://user-images.githubusercontent.com/22267870/83214872-d722d180-a18f-11ea-9c8f-7c8c3cd8808a.gif)

Just kiding! As @Jaex sai

> That image editor is actually RegionCaptureForm in codes, modified to act like image editor. So everything I add to region capture on screen annotation also benefits that mode. But this situation makes adding image editor related stuffs difficult. Because it is not coded from beginning to be image editor. So adding such features far more difficult because it is connected to region capture.

Might be it's real difficult because of how the source code "already" it is

**Comment by WhiteChairFromIkea** [comment]
Just voting up :)

**Comment by markusd1984** [comment]
Desperately need a ZOOM OUT function for screenshots with high resolutions to be able to highlight areas in one go rather then scroooooooolllllling

**Comment by gaia** [comment]
+1

**Comment by dmutsaers** [comment]
This would be a much welcomed enhancement/improvement :-)

**Comment by EpicLPer** [comment]
Also here to say I'd love to have this, the editor is really well made so far!

**Comment by ybekirov** [comment]
+1

Would also like this!

**Comment by ignaciobfp** [comment]
Would also like to see this feature added, the editor is surprisingly powerful and friendly to use, but I am missing the ability to zoom!

**Comment by Ronald-G-Andrade** [comment]
Please add this feature :(

**Comment by MHketbi** [comment]
i guess since it has been 4 years since op, then maybe close this ticket as refused/declined?

**Comment by WhiteChairFromIkea** [comment]
Sorry, (maybe I will get an answer because of many users in this thread): Maybe someone could recommend alternative for Sharex (freeware or opensource)? (Not having zoom functionality is really painfull, I would like to switch, but don't want to spend time trying out unknown apps).

**Comment by Davexx** [comment]
@WhiteChairFromIkea ,
See my comments and instructions from **March 9, 2020** and **May 7, 2020**.  
Windows 10 has vastly improved the Windows Snipping Tool making ShareX nearly obsolete, for most functions.

**Comment by MHketbi** [comment]
> 
> 
> Sorry, (maybe I will get an answer because of many users in this thread): Maybe someone could recommend alternative for Sharex (freeware or opensource)? (Not having zoom functionality is really painfull, I would like to switch, but don't want to spend time trying out unknown apps).

@WhiteChairFromIkea there are many photo editing software available check out this page
https://alternativeto.net/category/phots-and-graphics/photo-editing/

**Comment by markusd1984** [comment]
> > 3.5 years and nothing happend?
> > I'm waiting for this useful zoom feature too.
> > Thanks.
> 
> They created magnify instead 🤣🤣
> 
> > ![VqC2qhlH6x](https://user-images.githubusercontent.com/22267870/95704445-0e93a580-0c7b-11eb-85ea-4469e1f91434.gif)
> 
> This is another "solution" : ))
> 
> > This is an idea to zoom out your image. Howerver, you can't undo or zoom in again without losing image pixel : )
> > ![O7R8m98p1v](https://user-images.githubusercontent.com/22267870/83214872-d722d180-a18f-11ea-9c8f-7c8c3cd8808a.gif)
> 
> Just kiding! As @Jaex sai
> 
> > That image editor is actually RegionCaptureForm in codes, modified to act like image editor. So everything I add to region capture on screen annotation also benefits that mode. But this situation makes adding image editor related stuffs difficult. Because it is not coded from beginning to be image editor. So adding such features far more difficult because it is connected to region capture.
> 
> Might be it's real difficult because of how the source code "already" it is

fair enough, not only would it have to scale the image but all annotations, which is probably why it hasn't been implemented or ever will be, but would be good to have confirmation on this then all the requests in here. :)

**Comment by gustavotrott** [comment]
> 
> 
> @WhiteChairFromIkea ,
> See my comments and instructions from **March 9, 2020** and **May 7, 2020**.
> Windows 10 has vastly improved the Windows Snipping Tool, now called **Snip & Sketch** making ShareX nearly obsolete, for most functions. Access easily with WIN+SHIFT+S.

The tools available to draw in Windows Snipping Tool is very basic!! The only thing it is better than ShareX is the Zoom functionality.
ShareX have the best tools and is light at the same time! The Zoom In/Zoom Out is the only thing that is missing to be perfect!
I don't want to search for other capture program 😕

**Comment by WhiteChairFromIkea** [comment]
Is anyone working on zoom implementation? (I would rather use buggy version of a fork instead of official version without zoom function). Looked at the network graph, it seems no one is working on that :/

**Comment by Ronald-G-Andrade** [comment]
> 
> 
> Is anyone working on zoom implementation? (I would rather use buggy version of a fork instead of official version without zoom function). Looked at the network graph, it seems no one is working on that :/

😢

**Comment by k4zo** [comment]
4 years and still nothing, nice

**Comment by mturilin** [comment]
+1 to have a zoom feature. 

My use case:
- I needed to annotate my driving license to send to a customer support without some personal info
- I decided to user ShareX because this is the app that I use for annotation in general
- I opened the scanned file and it was too big because of the high resolution 
- I didn't have an option to reduce the image size except resizing the whole image. This was workable but far from elegant.

**Comment by aferrazlr** [comment]
+1 vote for zooming feature in ShareX Image Editor.

However my original request ( #5818 ) was to have an option for having the image always automatically resized to the window size. I believe this would be probably simpler from an UX perspective and cover +90% of the use cases described above.

**Comment by ivolake** [comment]
+1 to have a zoom feature.

**Comment by van1337** [comment]
+1

**Comment by stranno** [comment]
It's a pita to crop large images without a zoom feature. But, being 4 years already, I guess it is not that easy to do.

**Comment by vpr2k1** [comment]
What, there is still no function? I can't believe it (it's very sad, I used this program before and the lack of zoom made the use very inconvenient, otherwise it was good, then I decided to install it and was stunned by the lack of zoom (( how it's possible?

**Comment by kirillrocks** [comment]
Pals, requesting and complaining is easy, but if you read about this issue from the start, @Jaex said that ShareX doesn't have an image editor. And that is why it is very complicated(if even possible) to add a Zoom feature.

If ShareX had an image editor(and not an annotation tool acting like an image editor), they would add it a long time ago, as even @Jaex agreed it is a needed feature.

I am all about requesting features, but if I am feeling upset at the comments, image how @Jaex feels!

**Comment by WhiteChairFromIkea** [comment]
Pull requests are ignored by devs, how should contributors feel?

**Comment by kirillrocks** [comment]
> @WhiteChairFromIkea , See my comments and instructions from **March 9, 2020** and **May 7, 2020**. Windows 10 has vastly improved the Windows Snipping Tool, now called **Snip & Sketch** making ShareX nearly obsolete, for most functions. Access easily with WIN+SHIFT+S.

The Windows snipping tool lacks a lot of the essential features ShareX has. Workflows, destination integrations, color picker, image effects, and more options many use daily. If you are using ShareX just for the occasional screenshot, I guess even the default print-screen feature would suffice.

**Comment by vpr2k1** [comment]
> Pals, requesting and complaining is easy, but if you read about this issue from the start, @Jaex said that ShareX doesn't have an image editor. And that is why it is very complicated(if even possible) to add a Zoom feature.
> 
> If ShareX had an image editor(and not an annotation tool acting like an image editor), they would add it a long time ago, as even @Jaex agreed it is a needed feature.
> 
> I am all about requesting features, but if I am feeling upset at the comments, image how @Jaex feels!

WTF
people above wrote and threw off examples of simpler software with this function.

zoom = graphics editor? and the rest of the million functions and the built-in GRAPHIC EDITOR are ok, and the zoom is bad. what did you smoke bro?)

**Comment by mayurjobanputra** [comment]
> 

Bugshooting was my long-time favorite, until I moved to ShareX recently. I might go back to it.

**Comment by Lewis8379** [comment]
What a complete scandal that something as basic as zoom still hasn't been added since 2017. ShareX is dead to me. So disappointing.

**Comment by vpr2k1** [comment]
> What a complete scandal that something as basic as zoom still hasn't been added since 2017. ShareX is dead to me. So disappointing.

yes you absolutely right? thats sht annoying, but what is alternative?(

**Comment by EricTetz** [comment]
> @Jaex said that ShareX doesn't have an image editor. 

The problem is that this is an utterly nonsensical claim. _Of course_ ShareX has an image editor. It's a tool, titled "Image Editor", that takes an image, lets you make changes to it, then save a new, update image. It takes some serious mental gymnastics to decide that's _not_ an image editor.

This assertion bugged me so much that I pulled down the source and took a look. It took three lines of code to get basic scaling in the editor. Here it is, zooming in/out on the code change with some annotations added:

![](https://i.imgur.com/n3M2tzv.gif)

Actually implementing this for real would require more work. You need to translate mouse input, decide if you want to scale annotation handles (its own can of worms), perhaps add a modifier key (since it clashes with the magnifier), update documentation, so on and so forth. 

But there's no reason in principle that it can't be added. It's certainly not that the image editor is not an image editor, because... of course it is.

**Comment by derekantrican** [comment]
@EricTetz back when I had a bit more time (literally like over a year ago) I was looking at a way to fix this issue. Can you send me a fork of this repo with your changes?

**Comment by EricTetz** [comment]
The changes are in the screen cap. Really just the three lines. Will fork it later, do a little preliminary work, then send it your way.

**Comment by L1Q** [comment]
How is your performance when doing this? Enable "Show FPS" in the ⚙ menu.

**Comment by WhiteChairFromIkea** [comment]
Please post here a message when your fork w8ll be rpretty stable, I will use yours 👍

**Comment by WhiteChairFromIkea** [comment]
Tkank you!

**Comment by Jaex** [comment]
> The problem is that this is an utterly nonsensical claim. Of course ShareX has an image editor. It's a tool, titled "Image Editor", that takes an image, lets you make changes to it, then save a new, update image. It takes some serious mental gymnastics to decide that's not an image editor.

> This assertion bugged me so much that I pulled down the source and took a look.

> It's certainly not that the image editor is not an image editor, because... of course it is.

Maybe instead of implicitly insulting me you should try to understand why I repeatedly mentioned about why it is not exactly works like other image editors.

ShareX didn't had image editor before, but it had region capture with on screen annotation support. After tons of workarounds I made it windowed region capture but named it "Image editor" in tools menu to not confuse people. ShareX region capture rendering works different than other image editors because its rendering system works similar to game engines. It non stop renders in loop. In normal image editors your changes are directly applied to cache image and that cache image is rendered only once unless there is need for changes. So doing zooming etc. wouldn't have any performance issues in properly written image editor. That's why I told before it would need rewriting whole thing, because I must change whole rendering system but current rendering system works well for region capture, just not suitable for image editor.

So my concern was not about not being able to implement zoom support but it is about performance. Even now some people have very low FPS in region capture. So adding zoom support would make it even more unusable for those people.

I tried your codes now and this is the result:

Without zoom 385 FPS: https://jaex.getsharex.com/2022/01/ShareX_BHlIZwtOJo.png
With x1.1 zoom 44 FPS: https://jaex.getsharex.com/2022/01/ShareX_6PPrcLRaCb.png

So with your 3 lines of codes FPS dropped more than 8 times. If anyone be able to implement zoom support without any performance issues and side effects then of course I would gladly accept that pull request.

**Comment by EricTetz** [comment]
> my concern was not about not being able to implement zoom support

You didn't say that. I was responding to the assertion that an image editor is not an image editor, because of implementation details.

> I tried your codes now and this is the result:
> Without zoom 385 FPS: https://jaex.getsharex.com/2022/01/ShareX_BHlIZwtOJo.png
> With x1.1 zoom 44 FPS: https://jaex.getsharex.com/2022/01/ShareX_6PPrcLRaCb.png

1. Grab a screenshot.
1. Hit "Save" or "Save As". Save the current screenshot to disk. Leave editor open.
2. Click "Image(File)". Add the screenshot you just saved back into the current editor as a shape.
3. Click "Crop" and crop the original image down to a few pixels.
4. Watch most if not all of your performance come back, rendering exactly the same pixels as before, even while zoomed.

So the issue is not that GDI can't handle a scale transform on a modern computer, there's something in rendering is broken that is demonstrably fixable. I was going to take a look at it, but I think I'm just gonna back away slowly.

**Comment by EricTetz** [comment]
> Can you send me a fork of this repo with your changes?

@derekantrican Here you go: https://github.com/erictetz/sharex  Basic strategy is in place.

[Here's a video of zooming in action.](https://i.imgur.com/iMSq97w.mp4)  One line change to get most of the performance back. Zooming affects framerate significantly less than the magnifier.

Overall changes are minimal. Mostly scaling mouse coordinates for input.

TODO: make magnifier and info bar use correct coordinates, and find anything else that's not using the right coordinates. If you need help, let me know.

**Comment by Davexx** [comment]
@derekantrican @EricTetz  Heroes of the village!  THANK YOU!

**Comment by WhiteChairFromIkea** [comment]
@derekantrican @EricTetz, Thank you, this will be the version I am using from now on! It just works! :))

**Comment by derekantrican** [comment]
To be clear, I haven't done anything - it's been all @EricTetz ! I just asked for the fork.

If I have the time, I'd like to look at it, maybe help out in the fork where I can, then make a PR for this main repo but we'll see if/when I have the time for that.

**Comment by nepomuc** [comment]
I don't know how to compile @EricTetz 's version for myself :( Can anyone help? I also checked https://github.com/EricTetz/ShareX/releases but it's empty.

**Comment by WhiteChairFromIkea** [comment]
> I don't know how to compile @EricTetz 's version for myself :( Can anyone help? I also checked https://github.com/EricTetz/ShareX/releases but it's empty.

Download ms visual studio 2019 community edition (it is free for personal use), install. Open *.sln file (vs stidio solution file), it will be opened by visual studio. You will probably get a prompt to download "net framework v4.8 developer edition", search and download it from microsoft, when reopen sln file. Once opened, in the upper toolbar select "release" in dropdown (near green "play" button), and press green "play" button. After few minutes, dlls and exe files will be built into sherex\bin\release folder. Make a backup of your existing sharex installation, drag and drop non config (dlls, exe files) from bin folder to current installation of your sharex (replace existing executables).

Or, maybe I or someone else will put these binaries as a release zip, but no timeframe is known..

Also, I just tried zoomin\out function with a touchpad once, haven't time to use this version daily.

**Comment by EricTetz** [comment]
I thought derekantrican was chomping at the bit to contribute, so I stopped myself from finishing it. If that's not the case, I can finish it this weekend (maybe, have a gig, will be busy).

**Comment by derekantrican** [comment]
@EricTetz Oh, I would love to - just literally don't have the time. Don't hold back on my account :)

**Comment by EricTetz** [comment]
Had time to work on it today. [Pull request created.](https://github.com/ShareX/ShareX/pull/6084)

Never used GIT before, so hopefully I did it correctly.

![ZoomDemo](https://user-images.githubusercontent.com/929211/153084931-5466572e-9244-4af4-b51f-2a0a685ce17c.gif)

**Comment by Ronald-G-Andrade** [comment]
> Had time to work on it today. [Pull request created.](https://github.com/ShareX/ShareX/pull/6084)
> 
> Never used GIT before, so hopefully I did it correctly.
> 
> ![ZoomDemo](https://user-images.githubusercontent.com/929211/153084931-5466572e-9244-4af4-b51f-2a0a685ce17c.gif)

EXCELLENT dear friend 😄 

Please don't forget to add a keyboard shortcut to reset zoom to 100 %
The zoom status in the title bar is perfect 👍🏻

**Comment by WhiteChairFromIkea** [comment]
@EricTetz, I propose for you to open "issues" section at your fork. I would like a feature request: checkbox called "Zoom to fit on startup" in image editor settings (reduce drawing canvas to become all visible upon image editor starts). I don't believe sharex team will merge your PR (otherwise could implement zoom functionality ages ago themselves..)...
